###################################################################################################
Date: Fri, 7 Dec 2018 15:42:24 +0000
From 32cfb60130f85f93e9506e9d929e5bd0154c7ee7 Mon Sep 17 00:00:00 2001
From: Ben Gray <ben.gray@sky.uk>
Subject: [PATCH] Remove autoconnect when device is removed.

Without this when a device is removed then it remains in the autoconnect
list and in the kernels whitelist.  Meaning it will automatically
reconnect and we can never get rid of the device.

Source: SKY
Upstream-Status: Pending
Signed-off-by: Tijo Thomas John <tijo.thomas@sky.uk>
###################################################################################################

Index: bluez-5.76/src/device.c
===================================================================
--- bluez-5.76.orig/src/device.c
+++ bluez-5.76/src/device.c
@@ -6062,6 +6062,8 @@ void btd_device_set_temporary(struct btd
 	if (temporary) {
 		if (device->bredr)
 			adapter_accept_list_remove(device->adapter, device);
+                else
+                        adapter_auto_connect_remove(device->adapter, device);
 		adapter_connect_list_remove(device->adapter, device);
 		if (device->auto_connect) {
 			device->disable_auto_connect = TRUE;
